src/bin/cargo_stable_mir_json.rs

```diff
@@ -0,0 +1,181 @@
+use std::env;
+
+use std::io::Write;
+use std::os::unix::fs::PermissionsExt;
+use std::path::{Path, PathBuf};
+
+use anyhow::{bail, Result};
+
+fn main() -> Result<()> {
+    let args: Vec<_> = env::args().collect();
+    if args.len() != 2 {
+        bail!("Usage: cargo run --bin cargo_stable_mir_json -- <PATH-TO-STABLE-MIR-JSON-REPO>")
+    }
+    let repo_path: PathBuf = args[1].clone().into();
+    if !repo_path.is_dir() {
+        bail!(
+            "Provided should be path to stable_mir_json repo, but {} is not a dir",
+            repo_path.display()
+        );
+    }
+
+    let hidden_dir = setup(repo_path)?;
+    record_ld_library_path(&hidden_dir)?;
+    add_run_script(&hidden_dir)
+}
+
+fn add_run_script(hidden_dir: &Path) -> Result<()> {
+    let run_script_path = hidden_dir.join("run.sh");
+    let mut run_script = std::fs::File::create(&run_script_path)?;
+    writeln!(run_script, "#!/bin/bash")?;
+    writeln!(run_script, "set -eu")?;
+    writeln!(run_script)?;
+    writeln!(
+        run_script,
+        "export LD_LIBRARY_PATH=$(cat ~/.stable_mir_json/ld_library_path)"
+    )?;
+    writeln!(
+        run_script,
+        "exec \"/home/daniel/.stable_mir_json/debug/stable_mir_json\" \"$@\""
```

Comment:
Maybe use `$HOME` here?
```suggestion
        "exec \"$HOME/.stable_mir_json/debug/stable_mir_json\" \"$@\""
```